modules/navier_stokes/src/fvkernels/INSFVMomentumDiffusion.C

```diff
@@ -30,28 +33,66 @@ INSFVMomentumDiffusion::validParams()
                              "Switch that can select face interpolation method for the viscosity.");
 
   params.set<unsigned short>("ghost_layers") = 2;
+  params.addParam<bool>(
+      "complete_expansion",
+      false,
+      "Boolean parameter to use complete momentum expansion is the diffusion term.");
+  params.addCoupledVar("u", "The velocity in the x direction.");
+  params.addCoupledVar("v", "The velocity in the y direction.");
+  params.addCoupledVar("w", "The velocity in the z direction.");
```

Comment:
Add these as functors. More flexible and by adding as functors you won't trigger the "legacy" FV quadrature point evaluations that you get through the `Coupleable` interface

Reply:
Thanks. Changed to functors